GameServerLib/Scripting/CSharp/CSharpScriptEngine.cs

```diff
@@ -159,6 +159,34 @@ public T GetStaticMethod<T>(string scriptNamespace, string scriptClass, string s
             return default;
         }
 
+        /// <summary>
+        /// Grabs the current compiler assembly and creates a script object given a script namespace and class name.
+        /// </summary>
+        public static T CreateObjectStatic<T>(string scriptNamespace, string scriptClass)
+        {
+            var assembly = AppDomain.CurrentDomain.GetAssemblies().FirstOrDefault(a => a.GetName().Name == _assemblyName);
+            if (assembly == null)
+            {
+                return default;
+            }
+
+            scriptNamespace = scriptNamespace.Replace(" ", "_");
+            scriptClass = scriptClass.Replace(" ", "_");
+            string fullClassName = scriptNamespace + "." + scriptClass;
+
+            if (assembly.GetTypes().Any(t => t.FullName == fullClassName))
```

Comment:
Is it possible to do something like this here?
```csharp
var type = assembly.GetType(fullClassName, throwOnError: false);
if (type != null)
{
    return (T)Activator.CreateInstance(type);
}
```

Reply:
Logger is unavailable as it must be referenced.

Reply:
However, I can simply do `LoggerProvider.GetLogger().Warn` instead, although it'd mean making a reference every call.

Reply:
I just copied this line from the `CreateObject` function. The main thing that interested me was that it was possible to get rid of the lambda and there was no error message. To be honest, I still don't understand why the `CreateObjectStatic` function is needed

Reply:
Typically the project has been known to throw around references to low level classes such as Game and ScriptEngine quite a lot, and in the future it would be preferable to remove this necessity. This is an example of how it could be done.